shell/platform/embedder/embedder.h

```diff
@@ -562,6 +588,157 @@ typedef struct {
   const FlutterTaskRunnerDescription* platform_task_runner;
 } FlutterCustomTaskRunners;
 
+typedef struct {
+  // The type of the OpenGL backing store. Currently, it can either be a texture
+  // or a framebuffer.
+  FlutterOpenGLTargetType type;
+  union {
+    // A texture for Flutter to render into.
+    FlutterOpenGLTexture texture;
+    // A framebuffer for Flutter to render into. The embedder must ensure that
+    // the framebuffer is complete.
+    FlutterOpenGLFramebuffer framebuffer;
+  };
+} FlutterOpenGLBackingStore;
```

Comment:
Hmm, I see your point. But I am on the fence on this one. Since the addition of render targets for Vulkan & Metal will increase the backing store types, I'd rather just filter away entire target type classes in once shot in the future instead of adding client rendering API/target type combinations to this single enum.